src/ck-core/ckfutures.h

```diff
@@ -75,7 +78,7 @@ namespace ck {
       return std::move(holder.t);
     }
 
-    void set(const T &value) {
+    void set(const T &value) const {
```

Comment:
I'm not sure about how semantically valid it is for a setter to be const qualified. I suspect it only compiles because the `future` class and `CkFuture` struct themselves don't contain any data that changes during this operation.

Maybe there should be a discussion of what exactly it means for a future to be const-correct, beyond just letting certain code that uses the `const` keyword compile.

Reply:
See also: https://isocpp.org/wiki/faq/const-correctness#logical-vs-physical-const

Reply:
Just realized that set should not be `const` because it would then allow `const ck::future` to set value which completely nullifies the point of not allowing mutation with `const` objects. Apologies.